setup_vkd3d.sh

```diff
@@ -0,0 +1,191 @@
+#!/bin/bash
+
+# default directories
+vkd3d_lib32=${vkd3d_lib32:-"x86"}
+vkd3d_lib64=${vkd3d_lib64:-"x64"}
+
+# figure out where we are
+basedir=$(dirname "$(readlink -f $0)")
```

Comment:
Quote `$0` to prevent word splitting.
```suggestion
basedir=$(dirname "$(readlink -f "$0")")
```

Reply:
This doesn't make any difference on my end on folders with spaces.

Reply:
It certainly does on mine…

```
$ install -D /dev/stdin '/tmp/name with spaces/test.sh' <<'EOF'
#!/bin/bash
basedir=$(dirname "$(readlink -f $0)")
echo "First:  $basedir"
basedir=$(dirname "$(readlink -f "$0")")
echo "Second: $basedir"
EOF
$ '/tmp/name with spaces/test.sh'
First:  /tmp/name
/home/saancreed
Second: /tmp/name with spaces

Reply:
```
[joshua@lilypad frog with spaces]$ basedir=$(dirname "$(readlink -f $0)")
[joshua@lilypad frog with spaces]$ echo $basedir
/home/joshua/frog with spaces
[joshua@lilypad frog with spaces]$ 
```
And the script works fine if I run it in the "frog with spaces" folder.

Reply:
Not sure sure how your thing works anyway given the double quotes. (although I am not super up on my bash)

Wouldn't you want `\"`?

Reply:
> Wouldn't you want `\"`?

No, just like you don't want `\"` around `$(readlink …)`. These quotes are used for escaping spaces in `$0`, and using `\"` would cause them to be passed to `readlink`, so it would see the following parameters:

* `-f`
* `"/home/joshua/frog`
* `with`
* `spaces"`

When in doubt, ask `shellcheck`.

```
In setup_vkd3d.sh line 8:
basedir=$(dirname "$(readlink -f $0)")
                                 ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
basedir=$(dirname "$(readlink -f "$0")")
```

Also, I have no idea why your shell _doesn't_ perform word splitting on `$0` like it should, but maybe Hans-Kristian or Philip will.

Reply:
I'm using the bash straight from Arch Linux, we all use Arch (I think) so I think it'd work for all of us.

I'll take another look at this later today -- shellcheck looks cool, never heard of it before.

Reply:
Resolved.